packages/core/src/StateNode.ts

```diff
@@ -349,7 +349,7 @@ export class StateNode<
       () =>
         getCandidates(
           this,
-          eventName,
+          eventName as string,
```

Comment:
I've checked locally and it looks like this assertion is redundant
```suggestion
          eventName,
```